vision/src/autogluon/vision/configs/presets_configs.py

```diff
@@ -0,0 +1,184 @@
+        }
+    },
+)
+
+# Dictionary of preset fit() parameter configurations for ObjectDetector.
+preset_object_detector = dict(
+    # Best predictive accuracy with little consideration to inference time or model size. Achieve even better results by specifying a large time_limit value.
+    # Recommended for applications that benefit from the best possible model accuracy.
+    best_quality={
+        'hyperparameters': {
+            'transfer': Categorical('faster_rcnn_fpn_resnet101_v1b_coco'),
+            'lr': Real(1e-3, 1e-5, log=True),
+            'batch_size': Categorical(4, 8),
+            'epochs': 30
+            },
+        'hyperparameter_tune_kwargs': {
+            'num_trials': 128,
+            'search_strategy': 'bayesopt',
+            'max_reward': 1.0
+        }
+    },
+
+    # Good predictive accuracy with fast inference.
+    # Recommended for applications that require reasonable inference speed and/or model size.
+    good_quality_fast_inference={
+        'hyperparameters': {
+            'transfer': Categorical('ssd_512_resnet50_v1_coco',
+                                    'yolo3_darknet53_coco',
+                                    'center_net_resnet50_v1b_coco'),
+            'lr': Real(1e-2, 1e-5, log=True),
+            'batch_size': Categorical(8, 16, 32, 64),
+            'epochs': 50
+            },
+        'hyperparameter_tune_kwargs': {
+            'num_trials': 512,
+            'search_strategy': 'bayesopt',
+            'max_reward': 1.0
+        }
+    },
+
+    # Medium predictive accuracy with very fast inference and very fast training time. 
+    # This is the default preset in AutoGluon, but should generally only be used for quick prototyping.
+    medium_quality_faster_train={
+        'hyperparameters': {
+            'transfer': Categorical('ssd_512_resnet50_v1_coco'),
+            'lr': 0.01,
+            'batch_size': Categorical(8, 16),
+            'epochs': 30
+            },
+        'hyperparameter_tune_kwargs': {
+            'num_trials': 16,
+            'search_strategy': 'random',
+            'max_reward': 0.85
+        }
+    },
+
+    # Medium predictive accuracy with very fast inference.
+    # Comparing with `medium_quality_faster_train` it uses faster model but explores more hyperparameters.
+    medium_quality_faster_inference={
+        'hyperparameters': {
+            'transfer': Categorical('center_net_resnet18_v1b_coco', 'yolo3_mobilenet1.0_coco'),
+            'lr': Categorical(0.01, 0.005, 0.001),
+            'batch_size': Categorical(32, 64, 128),
+            'epochs': Categorical(30, 50),
+            },
+        'hyperparameter_tune_kwargs': {
+            'num_trials': 32,
+            'search_strategy': 'bayesopt',
+            'max_reward': 0.9
+        }
+    },
+
+)
+
+_PRESET_DICTS = {
+    'image_predictor': preset_image_predictor, 
+    'object_detector': preset_object_detector,
+}
+
+
+def set_presets(preset_name, *args, **kwargs):
```

Comment:
Do you think tabular's preset functionality is general enough to move to core so we don't have to copy paste these functions?

Reply:
moving to core is nice, see my modifications, I prefer a named tuple or mapping so that I won't need to import specific preset dict from ag.core